test/arbitrum/JPEGzMainnetOracle.t.sol

```diff
@@ -0,0 +1,505 @@
+    assertEq(liquidationPenalty, ethVault.liquidationPenalty());
+    assertEq(address(jpegzOracle), address(ethVault.tcapOracle()));
+    assertEq(address(jpegz), address(ethVault.TCAPToken()));
+    assertEq(address(weth), address(ethVault.collateralContract()));
+    assertEq(address(ethOracle), address(ethVault.collateralPriceOracle()));
+    assertEq(address(ethOracle), address(ethVault.ETHPriceOracle()));
+    assertEq(treasury, ethVault.treasury());
+    assertEq(0 ether, ethVault.minimumTCAP());
+  }
+
+  function testConstructor_ShouldRevert_WhenBurnFeeIsHigh(uint256 _burnFee)
+    public
+  {
+    if (_burnFee > 1000) {
+      vm.expectRevert("VaultHandler::constructor: fee higher than MAX_FEE");
+    }
+
+    ETHVaultHandler vault = new ETHVaultHandler(
+      orchestrator,
+      divisor,
+      ratio,
+      _burnFee,
+      mintFee,
+      liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0
+    );
+
+    if (!(_burnFee > 1000)) {
+      assertEq(_burnFee, vault.burnFee());
+    }
+  }
+
+  function testConstructor_ShouldRevert_WhenMintFeeIsHigh(uint256 _mintFee)
+    public
+  {
+    vm.assume(_mintFee > 1000);
+
+    vm.expectRevert("VaultHandler::constructor: fee higher than MAX_FEE");
+    new ETHVaultHandler(
+      orchestrator,
+      divisor,
+      ratio,
+      burnFee,
+      _mintFee,
+      liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0
+    );
+  }
+
+  function testConstructor_ShouldRevert_WhenLiquidationPenaltyIsHigh(
+    uint256 _liquidationPenalty,
+    uint256 _ratio
+  ) public {
+    if (_liquidationPenalty + 100 < _liquidationPenalty) {
+      return;
+    }
+
+    if ((_liquidationPenalty + 100) >= _ratio) {
+      vm.expectRevert(
+        "VaultHandler::constructor: liquidation penalty too high"
+      );
+    }
+
+    ETHVaultHandler vault = new ETHVaultHandler(
+      orchestrator,
+      divisor,
+      _ratio,
+      burnFee,
+      mintFee,
+      _liquidationPenalty,
+      address(jpegzOracle),
+      jpegz,
+      address(weth),
+      address(ethOracle),
+      address(ethOracle),
+      treasury,
+      0
+    );
+
+    if (!((_liquidationPenalty + 100) >= _ratio)) {
+      assertEq(_liquidationPenalty, vault.liquidationPenalty());
+    }
+  }
+
+  function testSetLiquidationPenalty_ShouldUpdateValue(
+    uint256 _liquidationPenalty
+  ) public {
+    if (_liquidationPenalty + 100 < _liquidationPenalty) {
+      return;
+    }
+
+    if ((_liquidationPenalty + 100) >= ratio) {
+      vm.expectRevert(
+        "VaultHandler::setLiquidationPenalty: liquidation penalty too high"
+      );
+    }
+
+    orchestrator.setLiquidationPenalty(ethVault, _liquidationPenalty);
+  }
+
+  function testShouldUpdateRatio(uint256 _ratio) public {
+    if (_ratio < 100) {
```

Comment:
can we do `vm.assume((_ratio < 100) &&(ethVault.liquidationPenalty() + 100 >= _ratio) )`?